ouroboros-consensus/tools/db-convert/Main.hs

```diff
@@ -0,0 +1,275 @@
+
+  parseRecord = fmap getOnly parseRecord
+
+instance ParseField UTCTime
+
+instance ParseFields UTCTime
+
+instance ParseRecord UTCTime where
+
+  parseRecord = fmap getOnly parseRecord
+
+instance ParseField (Hash CB.Raw) where
+
+  readField = Options.eitherReader (first Text.unpack . decodeAbstractHash . Text.pack)
+
+instance ParseFields (Hash CB.Raw)
+
+instance ParseRecord (Hash CB.Raw) where
+
+  parseRecord = fmap getOnly parseRecord
+
+data Args w
+  = Convert
+      { epochDir :: w ::: Path Abs Dir <?> "Path to the directory containing old epoch files"
+      , dbDir :: w ::: Path Abs Dir <?> "Path to the new database directory"
+      , epochSlots :: w ::: Word64 <?> "Slots per epoch"
+      }
+  | Validate
+      { dbDir :: w ::: Path Abs Dir <?> "Path to the new database directory"
+      , configFile :: w ::: Path Abs File <?> "Configuration file (e.g. mainnet-genesis.json)"
+      , systemStart :: w ::: Maybe UTCTime <?> "System start time"
+      , requiresNetworkMagic :: w ::: Bool <?> "Expecto patronum?"
+      , genesisHash :: w ::: Hash CB.Raw <?> "Expected genesis hash"
+      , verbose :: w ::: Bool <?> "Enable verbose logging"
+      }
+  deriving (Generic)
+
+instance ParseRecord (Args Wrapped)
+
+deriving instance Show (Args Unwrapped)
+
+data ValidationError
+  = MkConfigError CC.Genesis.ConfigurationError
+  | InitialChainStateError Scheduling.Error
+  deriving (Show, Typeable)
+
+instance Exception ValidationError
+
+main :: IO ()
+main = do
+  (cmd :: Args Unwrapped) <- unwrapRecord "Byron DB converter"
+  case cmd of
+    Convert {epochDir, dbDir, epochSlots} -> do
+      (_, files) <- listDir $ epochDir
+      let epochFiles = filter (\f -> fileExtension f == ".epoch") files
+      putStrLn $ "Writing to " ++ show dbDir
+      for_ epochFiles $ \f -> do
+        putStrLn $ "Converting file " ++ show f
+        convertEpochFile (EpochSlots epochSlots) f dbDir
+    Validate
+      { dbDir
+      , configFile
+      , requiresNetworkMagic
+      , genesisHash
+      , verbose
+      } -> do
+        econfig <-
+          runExceptT $
+            CC.Genesis.mkConfigFromFile
+              (if requiresNetworkMagic then RequiresMagic else RequiresNoMagic)
+              (toFilePath configFile)
+              genesisHash
+        case econfig of
+          Left err -> throwIO $ MkConfigError err
+          Right config -> validateChainDb dbDir config verbose
+
+convertEpochFile
+  :: EpochSlots
+  -> Path Abs File -- ^ Input
+  -> Path Abs Dir -- ^ Ouput directory
+  -> IO (Either CC.ParseError ())
+convertEpochFile es inFile outDir =
+  let inStream = CC.parseEpochFileWithBoundary es (toFilePath inFile)
+      dbDir = outDir </> [reldir|immutable|]
+      encode = CB.serializeEncoding' . Byron.encodeByronBlock . Byron.ByronBlockOrEBB
+   in do
+        createDirIfMissing True dbDir
+        outFileName <-
+          parseRelFile $ "epoch-" <>
+            drop 2 (toFilePath (filename inFile))
```

Comment:
Add a comment to `drop 2` that this changes `XXXXX` to `XXX` or just a comment saying that the old format is `XXXXX.epoch` and the new one is `epoch-XXXXX.dat`